# Readme

links to spotify documents and otehr resources
    ° https://developer.spotify.com/documentation/web-api/reference/#/
    ° https://developer.spotify.com/dashboard/applications/
    ° https://www.base64encode.org/
    ° https://open.spotify.com/artist/721C5U5rM8J0jjq6IQuSBK
    ° https://developer.spotify.com/documentation/web-api/reference/#/
    


In [22]:
url_base = 'https://api.spotify.com/v1'

In [23]:
ep_artist = '/artists/{artistid}'

In [24]:
id_solstafir = '721C5U5rM8J0jjq6IQuSBK'
# se puede poner cualquier id de artista, este se obtiene de spotify en la barra de busqueda despues de /artists/xxxxxxxxx

In [25]:
url_base + ep_artist.format(artistid=id_solstafir)

'https://api.spotify.com/v1/artists/721C5U5rM8J0jjq6IQuSBK'

In [26]:
import requests

In [27]:
#hacemos una request
r = requests.get(url_base + ep_artist.format(artistid=id_solstafir))

In [28]:
r.status_code

401

#el 401 nos dice que no tenemos permisos para acceder a ese recurso

In [29]:
#revisamos el json que nos devuelve r
r.json()

{'error': {'status': 401, 'message': 'No token provided'}}

In [49]:
#accedemos a la url par aobtener el token
token_url = 'https://accounts.spotify.com/api/token'

In [50]:
params = {'grant_type': 'client_credentials'}

In [51]:
headrers = {'Authorization': 'Basic XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'}
#usar https://www.base64encode.org/ con client_id:client_secret (XXXXXX:YYYYY) de la pagina https://developer.spotify.com/dashboard/applications/

In [54]:
r = requests.post(token_url, data = params, headers = headrers)

In [55]:
r.status_code

200

In [ ]:
r.json()

In [57]:
#acceder al token directamente
token = r.json()['access_token']

In [ ]:
token

In [59]:
header = {'Authorization': 'Bearer {}'.format(token)}

In [60]:
r = requests.get(url_base + ep_artist.format(artistid=id_solstafir), headers = header)

In [61]:
r.status_code

200

In [62]:
r.json()

{'external_urls': {'spotify': 'https://open.spotify.com/artist/721C5U5rM8J0jjq6IQuSBK'},
 'followers': {'href': None, 'total': 123395},
 'genres': ['blackened screamo',
  'icelandic black metal',
  'icelandic metal',
  'icelandic pop',
  'pagan black metal',
  'post-metal',
  'viking black metal'],
 'href': 'https://api.spotify.com/v1/artists/721C5U5rM8J0jjq6IQuSBK',
 'id': '721C5U5rM8J0jjq6IQuSBK',
 'images': [{'height': 640,
   'url': 'https://i.scdn.co/image/ab6761610000e5ebb90a1aede7a026fbaed20013',
   'width': 640},
  {'height': 320,
   'url': 'https://i.scdn.co/image/ab67616100005174b90a1aede7a026fbaed20013',
   'width': 320},
  {'height': 160,
   'url': 'https://i.scdn.co/image/ab6761610000f178b90a1aede7a026fbaed20013',
   'width': 160}],
 'name': 'Sólstafir',
 'popularity': 38,
 'type': 'artist',
 'uri': 'spotify:artist:721C5U5rM8J0jjq6IQuSBK'}

In [63]:
#Hacemos una bsuqueda con una url

url_busqueda = 'https://api.spotify.com/v1/search'

In [64]:
search_params = {'q': 'solstafir', 'type': 'artist', 'market': 'IS'}

In [65]:
busqueda = requests.get(url_busqueda, params = search_params, headers = header)

In [66]:
busqueda.status_code

200

In [67]:
busqueda.json()

{'artists': {'href': 'https://api.spotify.com/v1/search?query=solstafir&type=artist&market=IS&offset=0&limit=20',
  'items': [{'external_urls': {'spotify': 'https://open.spotify.com/artist/721C5U5rM8J0jjq6IQuSBK'},
    'followers': {'href': None, 'total': 123395},
    'genres': ['blackened screamo',
     'icelandic black metal',
     'icelandic metal',
     'icelandic pop',
     'pagan black metal',
     'post-metal',
     'viking black metal'],
    'href': 'https://api.spotify.com/v1/artists/721C5U5rM8J0jjq6IQuSBK',
    'id': '721C5U5rM8J0jjq6IQuSBK',
    'images': [{'height': 640,
      'url': 'https://i.scdn.co/image/ab6761610000e5ebb90a1aede7a026fbaed20013',
      'width': 640},
     {'height': 320,
      'url': 'https://i.scdn.co/image/ab67616100005174b90a1aede7a026fbaed20013',
      'width': 320},
     {'height': 160,
      'url': 'https://i.scdn.co/image/ab6761610000f178b90a1aede7a026fbaed20013',
      'width': 160}],
    'name': 'Sólstafir',
    'popularity': 38,
    'type': 'a

In [69]:
import pandas as pd

In [70]:
# creamos un dataframe con el resultado del json

df = pd.DataFrame(busqueda.json()['artists']['items'])

In [71]:
df

external_urls  \
0  {'spotify': 'https://open.spotify.com/artist/7...   
1  {'spotify': 'https://open.spotify.com/artist/1...   
2  {'spotify': 'https://open.spotify.com/artist/3...   
3  {'spotify': 'https://open.spotify.com/artist/6...   

                         followers  \
0  {'href': None, 'total': 123395}   
1    {'href': None, 'total': 8597}   
2    {'href': None, 'total': 5060}   
3    {'href': None, 'total': 4464}   

                                              genres  \
0  [blackened screamo, icelandic black metal, ice...   
1  [icelandic folk, icelandic indie, icelandic po...   
2                      [nordic orchestra, orchestra]   
3  [classic icelandic pop, icelandic metal, icela...   

                                                href                      id  \
0  https://api.spotify.com/v1/artists/721C5U5rM8J...  721C5U5rM8J0jjq6IQuSBK   
1  https://api.spotify.com/v1/artists/1SKQlvND1dW...  1SKQlvND1dW9m2zUpdDWBb   
2  https://api.spotify.com/v1/artists/3RPu1pwXXrP...  3RPu1pwXXrPqamAp6OHZ8b   
3  https://api.spotify.com/v1/artists/6aWektyglfp...  6aWektyglfpO0k93ns3J1Z   

                                              images  \
0  [{'height': 640, 'url': 'https://i.scdn.co/ima...   
1  [{'height': 640, 'url': 'https://i.scdn.co/ima...   
2  [{'height': 640, 'url': 'https://i.scdn.co/ima...   
3  [{'height': 640, 'url': 'https://i.scdn.co/ima...   

                         name  popularity    type  \
0                   Sólstafir          38  artist   
1             Moses Hightower          28  artist   
2  Iceland Symphony Orchestra          27  artist   
3                       DIMMA          26  artist   

                                     uri  
0  spotify:artist:721C5U5rM8J0jjq6IQuSBK  
1  spotify:artist:1SKQlvND1dW9m2zUpdDWBb  
2  spotify:artist:3RPu1pwXXrPqamAp6OHZ8b  
3  spotify:artist:6aWektyglfpO0k93ns3J1Z

In [72]:
#ordenamos el dataframe por la columna popularity
df = df.sort_values(by = 'popularity', ascending = False)

In [73]:
df

external_urls  \
0  {'spotify': 'https://open.spotify.com/artist/7...   
1  {'spotify': 'https://open.spotify.com/artist/1...   
2  {'spotify': 'https://open.spotify.com/artist/3...   
3  {'spotify': 'https://open.spotify.com/artist/6...   

                         followers  \
0  {'href': None, 'total': 123395}   
1    {'href': None, 'total': 8597}   
2    {'href': None, 'total': 5060}   
3    {'href': None, 'total': 4464}   

                                              genres  \
0  [blackened screamo, icelandic black metal, ice...   
1  [icelandic folk, icelandic indie, icelandic po...   
2                      [nordic orchestra, orchestra]   
3  [classic icelandic pop, icelandic metal, icela...   

                                                href                      id  \
0  https://api.spotify.com/v1/artists/721C5U5rM8J...  721C5U5rM8J0jjq6IQuSBK   
1  https://api.spotify.com/v1/artists/1SKQlvND1dW...  1SKQlvND1dW9m2zUpdDWBb   
2  https://api.spotify.com/v1/artists/3RPu1pwXXrP...  3RPu1pwXXrPqamAp6OHZ8b   
3  https://api.spotify.com/v1/artists/6aWektyglfp...  6aWektyglfpO0k93ns3J1Z   

                                              images  \
0  [{'height': 640, 'url': 'https://i.scdn.co/ima...   
1  [{'height': 640, 'url': 'https://i.scdn.co/ima...   
2  [{'height': 640, 'url': 'https://i.scdn.co/ima...   
3  [{'height': 640, 'url': 'https://i.scdn.co/ima...   

                         name  popularity    type  \
0                   Sólstafir          38  artist   
1             Moses Hightower          28  artist   
2  Iceland Symphony Orchestra          27  artist   
3                       DIMMA          26  artist   

                                     uri  
0  spotify:artist:721C5U5rM8J0jjq6IQuSBK  
1  spotify:artist:1SKQlvND1dW9m2zUpdDWBb  
2  spotify:artist:3RPu1pwXXrPqamAp6OHZ8b  
3  spotify:artist:6aWektyglfpO0k93ns3J1Z

In [74]:
#el que nos interesa el solo el primer elemento
df = df.iloc[0]

In [75]:
df

external_urls    {'spotify': 'https://open.spotify.com/artist/7...
followers                          {'href': None, 'total': 123395}
genres           [blackened screamo, icelandic black metal, ice...
href             https://api.spotify.com/v1/artists/721C5U5rM8J...
id                                          721C5U5rM8J0jjq6IQuSBK
images           [{'height': 640, 'url': 'https://i.scdn.co/ima...
name                                                     Sólstafir
popularity                                                      38
type                                                        artist
uri                          spotify:artist:721C5U5rM8J0jjq6IQuSBK
Name: 0, dtype: object

In [87]:
#por ahora solo queremos quedarnos con el id del artista
id_sols = df['id']

In [88]:
id_sols

'721C5U5rM8J0jjq6IQuSBK'

# Obteniendo la discografia

In [78]:
import base64

In [193]:
#creamos una funcion para renovar el token 
def get_token(client_id, client_secret):
    encoded = base64.b64encode(bytes('{}:{}'.format(client_id, client_secret), 'utf-8'))
    params = {'grant_type': 'client_credentials'}
    headers = {'Authorization': 'Basic {}'.format(encoded.decode('utf-8'))}
    r = requests.post('https://accounts.spotify.com/api/token', data = params, headers = headers)
    if r.status_code != 200:
        print('Error en la request', r.json())
        return None
    else:
        print('Tken válido por {} segundos'.format(r.json()['expires_in']))
        return r.json()['access_token']        

In [194]:
client_id = 'XXXXXXXXXXXXXXXXXXXXXXXXX'
client_secret = 'XXXXXXXXXXXXX'

In [195]:
#pobamos las funcion anterior
token = get_token(clien_id, client_secret)

Tken válido por 3600 segundos


In [173]:
header = {'Authorization': 'Bearer {}'.format(token)}

In [174]:
id_sols

'721C5U5rM8J0jjq6IQuSBK'

In [93]:
ep_albums = '/artists/{id}/albums'

In [94]:
url_base + ep_albums.format(id=id_sols)

'https://api.spotify.com/v1/artists/721C5U5rM8J0jjq6IQuSBK/albums'

In [95]:
params = {'country': 'IS'}

In [96]:
albums_sols = requests.get(url_base + ep_albums.format(id=id_sols), params = params, headers = header)

In [97]:
albums_sols.status_code

200

In [99]:
albums_sols.json()

{'href': 'https://api.spotify.com/v1/artists/721C5U5rM8J0jjq6IQuSBK/albums?offset=0&limit=20&include_groups=album,single,compilation,appears_on&market=IS',
 'items': [{'album_group': 'album',
   'album_type': 'album',
   'artists': [{'external_urls': {'spotify': 'https://open.spotify.com/artist/721C5U5rM8J0jjq6IQuSBK'},
     'href': 'https://api.spotify.com/v1/artists/721C5U5rM8J0jjq6IQuSBK',
     'id': '721C5U5rM8J0jjq6IQuSBK',
     'name': 'Sólstafir',
     'type': 'artist',
     'uri': 'spotify:artist:721C5U5rM8J0jjq6IQuSBK'}],
   'external_urls': {'spotify': 'https://open.spotify.com/album/7Gz2aaP3iC3eEzoNaMno4E'},
   'href': 'https://api.spotify.com/v1/albums/7Gz2aaP3iC3eEzoNaMno4E',
   'id': '7Gz2aaP3iC3eEzoNaMno4E',
   'images': [{'height': 640,
     'url': 'https://i.scdn.co/image/ab67616d0000b273ca6f61e4b859c3e4fdec46da',
     'width': 640},
    {'height': 300,
     'url': 'https://i.scdn.co/image/ab67616d00001e02ca6f61e4b859c3e4fdec46da',
     'width': 300},
    {'height': 64

In [100]:
#nos quedamos solo con los items
albums_sols.json()['items']

[{'album_group': 'album',
  'album_type': 'album',
  'artists': [{'external_urls': {'spotify': 'https://open.spotify.com/artist/721C5U5rM8J0jjq6IQuSBK'},
    'href': 'https://api.spotify.com/v1/artists/721C5U5rM8J0jjq6IQuSBK',
    'id': '721C5U5rM8J0jjq6IQuSBK',
    'name': 'Sólstafir',
    'type': 'artist',
    'uri': 'spotify:artist:721C5U5rM8J0jjq6IQuSBK'}],
  'external_urls': {'spotify': 'https://open.spotify.com/album/7Gz2aaP3iC3eEzoNaMno4E'},
  'href': 'https://api.spotify.com/v1/albums/7Gz2aaP3iC3eEzoNaMno4E',
  'id': '7Gz2aaP3iC3eEzoNaMno4E',
  'images': [{'height': 640,
    'url': 'https://i.scdn.co/image/ab67616d0000b273ca6f61e4b859c3e4fdec46da',
    'width': 640},
   {'height': 300,
    'url': 'https://i.scdn.co/image/ab67616d00001e02ca6f61e4b859c3e4fdec46da',
    'width': 300},
   {'height': 64,
    'url': 'https://i.scdn.co/image/ab67616d00004851ca6f61e4b859c3e4fdec46da',
    'width': 64}],
  'name': 'Endless Twilight of Codependent Love',
  'release_date': '2020-11-06',
 

In [101]:
#vemos el primer album 
albums_sols.json()['items'][0]

{'album_group': 'album',
 'album_type': 'album',
 'artists': [{'external_urls': {'spotify': 'https://open.spotify.com/artist/721C5U5rM8J0jjq6IQuSBK'},
   'href': 'https://api.spotify.com/v1/artists/721C5U5rM8J0jjq6IQuSBK',
   'id': '721C5U5rM8J0jjq6IQuSBK',
   'name': 'Sólstafir',
   'type': 'artist',
   'uri': 'spotify:artist:721C5U5rM8J0jjq6IQuSBK'}],
 'external_urls': {'spotify': 'https://open.spotify.com/album/7Gz2aaP3iC3eEzoNaMno4E'},
 'href': 'https://api.spotify.com/v1/albums/7Gz2aaP3iC3eEzoNaMno4E',
 'id': '7Gz2aaP3iC3eEzoNaMno4E',
 'images': [{'height': 640,
   'url': 'https://i.scdn.co/image/ab67616d0000b273ca6f61e4b859c3e4fdec46da',
   'width': 640},
  {'height': 300,
   'url': 'https://i.scdn.co/image/ab67616d00001e02ca6f61e4b859c3e4fdec46da',
   'width': 300},
  {'height': 64,
   'url': 'https://i.scdn.co/image/ab67616d00004851ca6f61e4b859c3e4fdec46da',
   'width': 64}],
 'name': 'Endless Twilight of Codependent Love',
 'release_date': '2020-11-06',
 'release_date_precisio

In [123]:
#veamos el nombre el album de todos los items
albums_sols.json()['items'][0]['name']

'Endless Twilight of Codependent Love'

In [163]:
#  partir de album_sols.json()['items'] vamos a buscar para cada item los datos que nos interesan
#creamos un dataframe para guardar los datos

info_solstafir = pd.DataFrame()

for item in albums_sols.json()['items']:
    item_alb_id = []
    #guardamos todos los itemps en una lista
    item_alb_id.append(item['id'])
    item_alb_name = []
    item_alb_name.append(item['name'])
    item_alb_track_to = []
    item_alb_track_to.append(item['total_tracks'])
    item_alb_reali_date = []
    item_alb_reali_date.append(item['release_date'])
    #juntamos las listas con un for
    for i in item_alb_id:
        for j in item_alb_name:
            for k in item_alb_track_to:
                for l in item_alb_reali_date:
                    #guardamos i j k l en info_solstafir
                    info_solstafir = info_solstafir.append({'id': i, 'name': j, 'total_tracks': k, 'release_date': l}, ignore_index = True)


C:\Users\gmgar\AppData\Local\Temp\ipykernel_648\2097546196.py:22: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  info_solstafir = info_solstafir.append({'id': i, 'name': j, 'total_tracks': k, 'release_date': l}, ignore_index = True)
C:\Users\gmgar\AppData\Local\Temp\ipykernel_648\2097546196.py:22: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  info_solstafir = info_solstafir.append({'id': i, 'name': j, 'total_tracks': k, 'release_date': l}, ignore_index = True)
C:\Users\gmgar\AppData\Local\Temp\ipykernel_648\2097546196.py:22: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  info_solstafir = info_solstafir.append({'id': i, 'name': j, 'total_tracks': k, 'release_date': l}, ignore_index = True)
C:\Users\gmgar\AppData\Local\Temp\ipy

In [164]:
info_solstafir

id                                               name  \
0   7Gz2aaP3iC3eEzoNaMno4E               Endless Twilight of Codependent Love   
1   4y0ReoVsLvfwFLRX5CcBsi                               Berdreyminn (Deluxe)   
2   0HqcHMC54fcVF11WuN5fV2                                        Berdreyminn   
3   3jUZIR2RRcuddKrbetFftv                                               Ótta   
4   36yXQj7nPEk7jpvIXOcsJh              Í Blóði Og Anda (In Blood and Spirit)   
5   6F2tCttIOIJzvlK4f6SE40                                     Svartir Sandar   
6   1q1hSeqTm10Bq4lspbv6L8                                               Köld   
7   6yvVBPCNlF5aBco71FbHV0                          Masterpiece Of Bitterness   
8   44orwmOzYwPURoW2Yfnpfb               Endless Twilight of Codependent Love   
9   4xHJON5VxzZqwPQi2sStIx                                Her Fall from Grace   
10  0iKkEZpsHWqoF1O5ESG2fE                                            Drýsill   
11  19jdHiHUBEweipwYFE72aw                                             Akkeri   
12  3CWXaYYPZCt7oRLVpG3fzd                                       Silfur-Refur   
13  2osllE3L4eRmKweIM17JFi                                       Silfur-Refur   
14  1yZS7rZncjhdFdWFdxhQ6F                                           Bláfjall   
15  5HQmpOpAVPsExLfPpoSlxT                                            Ísafold   
16  1bPNWKXL9dsaOvn7Mj5fZT                              Fjara / Runaway Train   
17  1porB6RyxZKI0Llyf80V6K  Spirit in Black, Chapter Two (The Ultimate Met...   

    total_tracks release_date  
0              9   2020-11-06  
1             10   2017-05-26  
2              8   2017-05-26  
3              8   2014-09-02  
4             19   2013-11-11  
5             12   2012-11-11  
6              8   2009-01-21  
7              7   2005-12-27  
8              2   2020-11-06  
9              1   2020-10-02  
10             1   2020-09-04  
11             1   2020-08-06  
12             2   2017-04-28  
13             1   2017-04-21  
14             1   2017-03-24  
15             1   2017-03-03  
16             2   2014-02-04  
17            25   2015-11-13

In [165]:
album_ep = '/albums/{album_id}'
album_params = {'market': 'IS'}


In [167]:
#seleccionamos un id
bnw_id = '3jUZIR2RRcuddKrbetFftv'

In [175]:
bnw = requests.get(url_base + album_ep.format(album_id=bnw_id)+'/tracks', params = album_params, headers = header)

In [176]:
bnw

<Response [200]>

In [177]:
bnw.json()

{'href': 'https://api.spotify.com/v1/albums/3jUZIR2RRcuddKrbetFftv/tracks?offset=0&limit=20&market=IS',
 'items': [{'artists': [{'external_urls': {'spotify': 'https://open.spotify.com/artist/721C5U5rM8J0jjq6IQuSBK'},
     'href': 'https://api.spotify.com/v1/artists/721C5U5rM8J0jjq6IQuSBK',
     'id': '721C5U5rM8J0jjq6IQuSBK',
     'name': 'Sólstafir',
     'type': 'artist',
     'uri': 'spotify:artist:721C5U5rM8J0jjq6IQuSBK'}],
   'disc_number': 1,
   'duration_ms': 524000,
   'explicit': False,
   'external_urls': {'spotify': 'https://open.spotify.com/track/5vkjx5WBPbAK7y3yUS3UnA'},
   'href': 'https://api.spotify.com/v1/tracks/5vkjx5WBPbAK7y3yUS3UnA',
   'id': '5vkjx5WBPbAK7y3yUS3UnA',
   'is_local': False,
   'is_playable': True,
   'name': 'Lágnætti',
   'preview_url': 'https://p.scdn.co/mp3-preview/a497b8a9833d0cb88773c32a40649d43230c0633?cid=ab89c76a89c44e5ebf2bb8fcc829aed7',
   'track_number': 1,
   'type': 'track',
   'uri': 'spotify:track:5vkjx5WBPbAK7y3yUS3UnA'},
  {'artists'

In [179]:
#nos quedamos con el nombre de la cancion unicamente
bnw.json()['items'][0]['name']

'Lágnætti'

In [180]:
[(track['id'], track['name']) for track in bnw.json()['items']]

[('5vkjx5WBPbAK7y3yUS3UnA', 'Lágnætti'),
 ('2LJZE1yXy9SdglruufEDHG', 'Ótta'),
 ('612b1EM4Q90ff5nWcci9Xi', 'Rismál'),
 ('2LuYF8vX3LwZMRQiXm9w3W', 'Dagmál'),
 ('258ihU9EiR7oG2a9tra6gR', 'Miðdegi'),
 ('5yHIeYnj6yvavfyjoNhYyK', 'Nón'),
 ('32slaVU9hlnlI7UDXE8xMP', 'Miðaftann'),
 ('4aei6FwNGzOWs3tEWpRrFi', 'Náttmál')]

In [197]:
def obtener_discografia(artist_id, token, return_name=False, page_limit=50, country=None):
    url = f'https://api.spotify.com/v1/artists/{artist_id}/albums'
    header = {'Authorization': f'Bearer {token}'}
    params = {'limit': page_limit, 'market': country, 'offset': 0}

    lista_albumes = []
    r = requests.get(url, headers=header, params=params)
    if r.status_code != 200:
        print('Error en la request', r.json())
        return None
    if return_name:
        lista_albumes += [(item['id'], item['name']) for item in r.json()['items']]
    else:
        lista_albumes += [item['id'] for item in r.json()['items']]

    while r.json()['next']:
        r = requests.get(r.json()['next'], headers=header)
        if r.status_code != 200:
            return None
        else:
            print('Tken válido por {} segundos'.format(r.json()['expires_in']))
            return r.json()['access_token']

        if return_name:
            lista_albumes += [(item['id'], item['name']) for item in r.json()['items']]
    return lista_albumes

In [198]:
def obtener_tracks(album_id, token, return_name=False, page_limit=50, market=None):
    url = f'https://api.spotify.com/v1/albums/{album_id}/tracks'
    header = {'Authorization': f'Bearer {token}'}
    params = {'limit': page_limit, 'offset': 0, market: market}

    lista_tracks = []
    r = requests.get(url, headers=header, params=params)
    if r.status_code != 200:
        print('Error en la request', r.json())
        return None

    if return_name:
        lista_tracks += [(item['id'], item['name'])
                         for item in r.json()['items']]
    else:
        lista_tracks += [item['id'] for item in r.json()['items']]

    while r.json()['next']:
        r = requests.get(r.json()['next'], headers=header)
        if r.status_code != 200:
            return None
        else:
            print('Tken válido por {} segundos'.format(r.json()['expires_in']))
            return r.json()['access_token']

        if return_name:
            lista_tracks += [(item['id'], item['name'])
                             for item in r.json()['items']]
        else:
            lista_tracks += [item['id'] for item in r.json()['items']]

    return lista_tracks


In [199]:
for album in obtener_discografia(id_sols, token, return_name=True, country='IS'):
    print(album[1])
    for track in obtener_tracks(album[0], token, return_name=True, market='IS'):
        print('\t', track[1])

Endless Twilight of Codependent Love
	 Akkeri
	 Drýsill
	 Rökkur
	 Her Fall from Grace
	 Dionysus
	 Til Moldar
	 Alda Syndanna
	 Ör
	 Úlfur
Berdreyminn (Deluxe)
	 Silfur-Refur
	 Ísafold
	 Hula
	 Nárós
	 Hvít Sæng
	 Dýrafjörður
	 Ambátt
	 Bláfjall
	 Svart Blóð
	 Samband I Berlin
Berdreyminn
	 Silfur-Refur
	 Ísafold
	 Hula
	 Nárós
	 Hvít sæng
	 Dýrafjörður
	 Ambátt
	 Bláfjall
Ótta
	 Lágnætti
	 Ótta
	 Rismál
	 Dagmál
	 Miðdegi
	 Nón
	 Miðaftann
	 Náttmál
Í Blóði Og Anda (In Blood and Spirit)
	 Undir Jökli - Vetrarins Dauðu Sumarblom
	 Í Blóði og Anda
	 The Underworld Song
	 Tormentor
	 2000 Ár
	 Ei Við Munum Iðrast
	 Bitch in Black
	 Í Víking
	 Árstíðir Dauðans
	 Undir Jökli - 98 demo
	 Í Blóði og Anda - 98 demo
	 The Underworld Song - 98 Demo
	 Tormentor - 98 Demo
	 2000 Ar - 98 demo
	 Piano Song - 98 Demo
	 Ei Við Munum Iðrast - 97 demo
	 Bitch in Black - 97 Demo
	 Í Víking - 97 demo
	 Undirheimalagið - Rehearsal Recording
Svartir Sandar
	 Ljós í Stormi
	 Fjara
	 Þín Orð
	 Sjúki Skuggin

In [206]:
preview_url = 'https://p.scdn.co/mp3-preview/a497b8a9833d0cb88773c32a40649d43230c0633?cid=ab89c76a89c44e5ebf2bb8fcc829aed7'

In [207]:
preview = requests.get(preview_url)
preview

<Response [200]>

In [208]:
import IPython.display as ipd

In [209]:
ipd.Audio(preview.content)